### PR TITLE
Implemented assignment questions on generators

### PR DESCRIPTION
**Pull request by shravan-cc**
1. This PR introduces an iterable linked list with methods to add items and iterate through the list. Additionally, it includes a generator function to iterate through the linked list.
2. Included comprehensive test cases to validate the functionality of the linked list and generator. Tests cover item addition, iteration using the spread operator, and iteration using the generator function.

### CODE REVIEW

File: iterator.js [hunk 1]
```diff
@@ -23,3 +23,72 @@ export const alphabetIterator = {
...
+    data,
+    next: null,
+  };
+}
+/**
+ * Creates an iterable linked list with methods to add items and iterate through the list.
+ *
+ * @returns {Object} An iterable linked list object with methods `addItem` and an iterator.
+ */
+export function linkedListIterator() {
```
Comment (shripada) [inline iterator.js]: This is actually the LinkedList factory actually that returns a list and that list is iterable. Rename this as createLinkedList

File: iterator.js [hunk 2]
```diff
@@ -23,3 +23,72 @@ export const alphabetIterator = {
...
+  return {
+    head: null,
```
Comment (shripada) [inline iterator.js]: We can completely avoid this members, and move them up in the createLinkedList, this will avoid `this`

**Review by shripada — CHANGES_REQUESTED** [review]

File: iterator.js (hunk 1 above)
Comment (shravan-cc) [reply iterator.js]: Addressed the changes.

File: iterator.js (hunk 2 above)
Comment (shravan-cc) [reply iterator.js]: Addressed the changes.

**Review by shripada — APPROVED** [review]

**Pull request merged**